clang/utils/TableGen/ClangDiagnosticsEmitter.cpp

```diff
@@ -468,11 +442,9 @@ static StringRef getModifierName(ModifierType MT) {
     return "objcclass";
   case MT_ObjCInstance:
     return "objcinstance";
-  case MT_Unknown:
+  default:
```

Comment:
This should NOT happen here.  Changing this to 'default' changes the way our warnings would work.  This switches us from a 'covers all enum elements' to one that doesn't.  Our switch cleanliness should be (in priority order):


1- Covers ALL enum values, no default.  This way when adding a new enum value we get a warning that something is uncovered.  

2- Covers MOST enum values, but has the 'rest' in a list with an 'unreachable' under it, again, no default. This DOES result in an 'unreachable' twice (or a just a 'break' in the body), but it makes adding/removing elements much easier, as it keeps the warning from above.

3- Cover a 'partial' list, uses a 'default'.  This is somewhat error-prone, as it doesn't warn when adding a new element, but is acceptable.


In this case, your change brings us from 2 to 3, which we shouldn't do.

Reply:
Done now.